tensorflow/compiler/tf2tensorrt/segment/segment.cc

```diff
@@ -679,6 +680,96 @@ void AddSegmentForNode(const grappler::GraphProperties* graph_properties,
 
 }  // namespace
 
+string GenerateUnconversionReport(
+    std::map<string, std::map<string, int>>& unconverted_ops_map) {
+  // Fetch whether to print a detailed version of the TF-TRT conversion report.
+  bool show_detailed_conversion_report;
+  TF_CHECK_OK(ReadBoolFromEnvVar("TF_TRT_SHOW_DETAILED_REPORT",
+                                 /*default_value=*/false,
+                                 &show_detailed_conversion_report));
+
+  string unsupported_op_report =
+      StrCat("\n\n", string(80, '#'), "\n",
+             "TensorRT unsupported/unconverted OP Report:");
+  int total_unconverted_ops{0};
+
+  // Copying map data into a vector for descending sorting.
+  std::vector<std::pair<
+      std::string,    // OP Name
+      std::pair<int,  // Total Unconverted of a specific OP
+                std::vector<std::pair<string, int>>  // {Reason, Total for this
+                                                     // reason}
+                >>>
```

Comment:
Let's use type aliases to simplify the use of the types in the compare lambdas.

Reply:
Fixed